.github/pull_request_template.md

```diff
@@ -6,6 +6,19 @@
 
 <!-- List the affected areas (e.g wallet, browser, etc..) -->
 
+### Developer Checklist
+
+#### Tests & QA
+- [ ] I have created user stories for this feature/rework
+- [ ] I have written unit tests for status-go and QML changes
+- [ ] I have created a storybook page for all UI components and views
+- [ ] I have tested this PR on both storybook and the actual client (for UI changes)
+- [ ] I have followed the architecture guidelines: [Architecture guidelines](../architecture-guide.md)
+- [ ] For bug fixes, I have written a new unit test case to prevent reintroduction
+- [ ] I have performed side impact testing for changes in shared components
+- [ ] A reviewer has Tested the PR during review
+- [ ] Given enough time for a thorough review
+
```

Comment:
A little bit of feedback here:

- Keep checkbox labels short, it's too much processing to read all of them
  - Example: 
      - `Has unit tests` instead of `I have written unit tests for status-go and QML changes`
      - `Has user stories` instead of `I have created stories for this feature/rework`
      - `Has storybook page (create if not existing)` etc ...
- The `I have followed the architecture guidelines` not sure this adds a lot of value here. While some of the points above can act as reminders (don't forget to add tests!) this one is more blurry. 
- Maybe instead of `I have performed side impact testing...` Add a section that says something like:

**Side effects caused by these changes worth mentioning:**

- The `A reviewer has tested this PR during review` is something that should just be tackled by GitHub's "Request Review" feature, which implies the result in this checkbox automatically
- The `Given enough time for a thorough review` is far fro quantifiable. If you want to have this guaranteed, it probably makes more sense to agree on a rule like: "Leave PRs at least # days pending for review, don't merge before"